pd/src/components/ibc.rs

```diff
@@ -32,11 +50,98 @@ impl Component for IBCComponent {
         Ok(())
     }
 
-    async fn execute_tx(&mut self, _tx: &Transaction) -> Result<()> {
-        Ok(())
+    async fn execute_tx(&mut self, tx: &Transaction) -> Result<()> {
+        // handle client transactions
+        for action in tx.transaction_body.actions.iter() {
+            let ibc_action = match action {
+                Action::IBCAction(ibc_action) => Some(ibc_action),
+                _ => None,
+            };
+            if ibc_action.is_none() {
+                continue;
+            }
+
+            match &ibc_action.unwrap().action {
```

Comment:
Pre-filtering avoids the need for this `unwrap`, since the `action` would already be an IBC action